contracts/settling_game/S05_Wonders.cairo

```diff
@@ -0,0 +1,171 @@
+%lang starknet
+%builtins pedersen range_check bitwise
+from starkware.cairo.common.bitwise import bitwise_and
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.cairo.common.math import assert_nn_le, unsigned_div_rem, assert_not_zero
+from starkware.cairo.common.math_cmp import is_nn_le
+from starkware.cairo.common.hash_state import hash_init, hash_update, HashState
+from starkware.cairo.common.alloc import alloc
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.uint256 import Uint256, uint256_eq
+from starkware.cairo.common.pow import pow
+from contracts.settling_game.utils.general import scale
+from contracts.settling_game.utils.interfaces import IModuleController
+
+from contracts.settling_game.utils.game_structs import ResourceLevel
+
+from contracts.token.ERC20.interfaces.IERC20 import IERC20
+from contracts.token.ERC1155.interfaces.IERC1155 import IERC1155
+from contracts.settling_game.interfaces.realms_IERC721 import realms_IERC721
+
+from contracts.settling_game.utils.game_structs import RealmData
+from contracts.settling_game.utils.general import unpack_data
+
+# #### Module 5B ##########
+#                        #
+# Wonder Tax Pool State  #
+#                        #
+##########################
+
+# Will keep track of active wonder allocation
+
+@storage_var
+func controller_address() -> (address : felt):
+end
+
+@storage_var
+func epoch_claimed(address : felt) -> (epoch : felt):
+end
+
+@storage_var
+func total_wonders_staked(epoch : felt) -> (amount : felt):
```

Comment:
I'm not super familiar with the logic of the Wonder tax, can you please explain what is this supposed to store? It's a mapping between a epoch number and the amount of wonders staked in that epoch, right? So will the wonder holders have to restake on every epoch?

Reply:
You're understanding is right, for this i think the best way to keep track of the total wonders is by always check for the passing of an epoch in tax claims, and roll forward the previously total settled wonders. Ill push an implementation tonight